fuzz-tests/src/bin/decimal.rs

```diff
@@ -50,4 +55,49 @@ fuzz_target!(|decimal: OneDecimal| {
     let _ = decimal.0.checked_powi(decimal.2);
     let _ = decimal.0.checked_nth_root(decimal.3);
      */
-});
\ No newline at end of file
+}
+
+// Fuzzer entry points
+#[cfg(feature = "libfuzzer-sys")]
```

Comment:
Since we will be focusing more on fuzzing as an automated testing method I think that it might make more sense for us to invest sometime into a `fuzz` macro to lower the amount of boilerplate code required to write fuzz tests that utilize libFuzzer and afl. I imagine something like the following:

```rs
use fuzz_tests::fuzz;

fuzz!(|input: OneDecimal| {
    /* Fuzzing */
})
```

Would expand to:

```rs
#[cfg(feature = "libfuzzer-sys")]
libfuzzer_sys::fuzz_target!(|decimal: OneDecimal| {
    /* Fuzzing */
});

#[cfg(feature = "afl")]
fn main() {
    afl::fuzz!(|decimal: OneDecimal| {
        /* Fuzzing */
    });
}

#[cfg(feature = "simple-fuzzer")]
fn main() {
    fuzz_tests::fuzz!(|decimal: OneDecimal| {
        /* Fuzzing */
    });
}
```

This is nice because it removes the conditional `use` statements, conditional `main` functions and most `cfg` and `cfg_attr`s from the code we write and off-loads that to the macro.

Reply:
I love it!